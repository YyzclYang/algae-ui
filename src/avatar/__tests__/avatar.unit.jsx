import React from 'react';
import * as renderer from 'react-test-renderer';
import Avatar from '../index';

describe('Avatar', () => {
  it('渲染默认 Avatar', () => {
    const component = renderer.create(<Avatar />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染方形的 Avatar', () => {
    const component = renderer.create(<Avatar shape="square" />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染自定义尺寸的 Avatar', () => {
    const component = renderer
      .create(<Avatar size={48} shape="square" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染自定义图片的 Avatar', () => {
    const component = renderer
      .create(
        <Avatar
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          alt="头像"
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染自定义文案的 Avatar', () => {
    const component = renderer.create(<Avatar>User</Avatar>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('渲染自定义样式的 Avatar', () => {
    const component = renderer
      .create(
        <Avatar style={{ color: 'red', background: '#FADFC9' }}>U</Avatar>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
