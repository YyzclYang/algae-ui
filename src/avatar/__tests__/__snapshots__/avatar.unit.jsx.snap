// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar 渲染方形的 Avatar 1`] = `
<span
  className="algae-ui-avatar algae-ui-avatar-square"
  style={
    Object {
      "height": "32px",
      "width": "32px",
    }
  }
>
  <svg
    className="algae-ui-icon algae-ui-avatar-icon"
    fill="#FFFFFF"
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    <use
      xlinkHref="#avatar"
    />
  </svg>
</span>
`;

exports[`Avatar 渲染自定义图片的 Avatar 1`] = `
<span
  className="algae-ui-avatar algae-ui-avatar-circle"
  style={
    Object {
      "background": "transparent",
      "height": "32px",
      "width": "32px",
    }
  }
>
  <img
    alt="头像"
    className="algae-ui-avatar-image"
    src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
  />
</span>
`;

exports[`Avatar 渲染自定义尺寸的 Avatar 1`] = `
<span
  className="algae-ui-avatar algae-ui-avatar-square"
  style={
    Object {
      "height": "48px",
      "width": "48px",
    }
  }
>
  <svg
    className="algae-ui-icon algae-ui-avatar-icon"
    fill="#FFFFFF"
    style={
      Object {
        "fontSize": "21px",
      }
    }
  >
    <use
      xlinkHref="#avatar"
    />
  </svg>
</span>
`;

exports[`Avatar 渲染自定义文案的 Avatar 1`] = `
<span
  className="algae-ui-avatar algae-ui-avatar-circle"
  style={
    Object {
      "height": "32px",
      "width": "32px",
    }
  }
>
  User
</span>
`;

exports[`Avatar 渲染自定义样式的 Avatar 1`] = `
<span
  className="algae-ui-avatar algae-ui-avatar-circle"
  style={
    Object {
      "background": "#FADFC9",
      "color": "red",
      "height": "32px",
      "width": "32px",
    }
  }
>
  U
</span>
`;

exports[`Avatar 渲染默认 Avatar 1`] = `
<span
  className="algae-ui-avatar algae-ui-avatar-circle"
  style={
    Object {
      "height": "32px",
      "width": "32px",
    }
  }
>
  <svg
    className="algae-ui-icon algae-ui-avatar-icon"
    fill="#FFFFFF"
    style={
      Object {
        "fontSize": "14px",
      }
    }
  >
    <use
      xlinkHref="#avatar"
    />
  </svg>
</span>
`;
